src/components/Login/index.jsx

```diff
@@ -1,74 +1,42 @@
-import React, { useState, useEffect } from "react";
-import axios from "axios";
+import React, { useState, useRef } from "react";
+import api from "../../api";
 
-import "./index.css";
+import { useNavigate, Link } from "react-router-dom";
 
-function Login() {
-  const [user, setUser] = useState({
-    userId: "",
-    password: "",
-  });
+import "./index.css";
 
-  const handleUserInput = (e) => {
-    setUser({
-      ...user,
-      [e.target.name]: e.target.value,
-    });
-  };
+function Login({ setNickname }) {
+  const [loginError, setLoginError] = useState("");
+  const emailInputRef = useRef();
+  const pwInputRef = useRef();
+  const navigate = useNavigate();
 
   const onClickLogin = (event) => {
     event.preventDefault();
-    console.log("ID : ", user.userId);
-    console.log("PW : ", user.password);
-    axios
-      .post(
-        "/user/login",
-        {
-          userId: user.userId,
-          password: user.password,
-        },
-        {
-          headers: {
-            "Content-type": "application/json",
-          },
-        }
-      )
-      .then((res) => {
-        console.log(res);
-        console.log("res.data.userId :: ", res.data.userId);
-        console.log("res.data.msg :: ", res.data.msg);
-        if (res.data.code === 200) {
-          console.log(res.data);
-        } else if (res.data.code === 400) {
-          console.log(res.data);
-        }
-        document.location.href = "/movie";
-      })
-      .catch((error) => {
-        console.log(error);
-      });
-    setUser({ userId: "", password: "" });
-  };
+    const enteredEmail = emailInputRef.current.value;
+    const enteredPassword = pwInputRef.current.value;
 
-  useEffect(() => {
-    axios
-      .get("/user/login")
-      .then((res) => console.log(res))
-      .catch();
-  }, []);
+    api.postLogin(enteredEmail, enteredPassword, ({ nickname, error }) => {
+      console.log(nickname, error);
+      if (error) {
+        setLoginError("아이디 또는 비밀번호를 다시 확인해주세요.");
+      }
```

Comment:
에러가 났다면 더 이상 진행되지 않도록 막아주어야합니다.
지금 상황에서는 유저가 `"아이디 또는 비밀번호를 다시 확인해주세요."`를 보기전에 /movie 페이지로 redirect 될 것으로 보입니다.
```suggestion
      if (error) {
        setLoginError("아이디 또는 비밀번호를 다시 확인해주세요.");
        return;
      }
```

Reply:
큰일날뻔했네요 😮 수정했습니다 🙇‍♀️🙇‍♂️